lib/OpenQA/Shared/GruJob.pm

```diff
@@ -41,13 +41,15 @@ sub execute {
 
 sub _delete_gru {
     my ($self, $id) = @_;
-    my $gru = $self->minion->app->schema->resultset('GruTasks')->find($id);
+    my $gru = $self->minion->app->schema->resultset('GruTasks')->find($id)
+      or $self->app->log->error("Could not find GruTask '$id' in database");
```

Comment:
Wouldn't we get an error in our logs with this every time someone restarts a Minion job via the Minion dashboard or when a Minion job is enqueued using the command API (like our cleanup tasks)? Or are those cases irrelevant because then this custom class/package is not used at all?

Of course the same counts for the other function.

Reply:
I removed that commit.
That was before I found out that it might be valid to restart a job even if it had been running successfully and deleted the gru task entry.